optics-core/src/Optics/Internal/Optic/TypeLevel.hs

```diff
@@ -107,23 +107,20 @@ data IxEq i is js where
 -- foldr f (foldr f init xs) ys = foldr f init (ys ++ xs)
 --    where f = (->)
 -- @
-class AppendIndices xs ys where
-  appendIndices :: IxEq i (Curry xs (Curry ys i)) (Curry (Append xs ys) i)
+class AppendIndices xs ys ks | xs ys -> ks where
+  appendIndices :: IxEq i (Curry xs (Curry ys i)) (Curry ks i)
 
--- | If we know the second list is empty, we can pick the first list without
--- knowing anything about it, hence the instance is marked as INCOHERENT.
-instance {-# INCOHERENT #-} AppendIndices xs '[] where
```

Comment:
Oh, I thought that `FunctionalDependencies` would choke on this? Is `INCOHERENT` lifting that somehow?

Reply:
It's not INCOHERENT, it's just that the consistency check for fundeps is quite lenient.